contracts/yToken.sol

```diff
@@ -1,34 +1,27 @@
 // SPDX-License-Identifier: GPL-3.0
-pragma solidity ^0.6.12;
-pragma experimental ABIEncoderV2;
+pragma solidity 0.8.3;
 
 import {Address} from "@openzeppelin/contracts/utils/Address.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
-import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
-
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 import {VaultAPI, BaseWrapper} from "./BaseWrapper.sol";
 
 contract yToken is IERC20, BaseWrapper {
-    using SafeMath for uint256;
-
     mapping(address => mapping(address => uint256)) public override allowance;
 
-    constructor(address _token, address _registry) public BaseWrapper(_token, _registry) {}
+    constructor(address _token, address _registry) BaseWrapper(_token, _registry) {}
 
     function name() external view returns (string memory) {
-        VaultAPI _bestVault = bestVault();
-        return _bestVault.name();
+        return string(abi.encodePacked("Yearn ", IERC20Metadata(address(token)).name()));
     }
 
     function symbol() external view returns (string memory) {
-        VaultAPI _bestVault = bestVault();
-        return _bestVault.symbol();
+        return string(abi.encodePacked("y", IERC20Metadata(address(token)).symbol()));
     }
 
     function decimals() external view returns (uint256) {
-        VaultAPI _bestVault = bestVault();
-        return _bestVault.decimals();
+        return IERC20Metadata(address(token)).decimals();
     }
```

Comment:
Can these changes be brought out into another PR? Basically, define `IERC20Metadata` interface above and use that until "upgrading" to OZ version

Reply:
ok will do